mobify/platform-scaffold
# [IM] Product Variation Support

# [IM] Product Variation Support

**The SDK PR https://github.com/mobify/progressive-web-sdk/pull/715 needs to be merged & released before merging this scaffold branch**

 **JIRA**: https://mobify.atlassian.net/browse/WEBDATA-88

## Todos

- [x] `Update Cart` functionality on merlin's variants
- [x] `Add to Cart` functionality on merlin's variants
- [x] ~~Refactor to avoid changing `addToCart` command parameters~~
- [x] Fix runtypes for new optional variationCategory property
- [x] Ensure preselect behavior exists as it did previously on SFCC
- [x] Test for regressions across both connectors
- [x] Test for regressions on non-variant products


## Changes
- Captures the variation data from products on Merlin's potions
- Format data to match the variation schema on SFCC connector
- Update addToCart / submitCartForm to allow adding to cart
- Update `cart-product-list` to display selected variants in cart

## How to test-drive this PR

```
cd progressive-web-sdk
git checkout im-add-to-cart-variants
npm link
npm install
npm run dev:build
cd ../platform-scaffold/web
git checkout product-variants-on-merlins
npm link progressive-web-sdk
npm run dev
```

- Use Merlin's Connector
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- Add Eye of Newt with variations to cart
- Attempt updating the cart
- Add a different variation to cart
- Ensure both variants are showing up as separate cart items
